### PR TITLE
SW-13872 - Add config.php guide

### PR DESCRIPTION
**Pull request by taltholtmann**

### CODE REVIEW

File: source/developers-guide/shopware-config/index.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+menu_title: config.php settings
+menu_order: 45
+---
+ 
+<div class="toc-list"></div>
+
+## Introduction
+In this guide we will take a closer look at the configuration file `config.php`.
+The file is in the root folder of a shopware installation. Normally it is auto 
+generated during the installation prozess and filled with your database credentials.
```
Comment (mitelg) [inline source/developers-guide/shopware-config/index.md]: process

File: source/developers-guide/shopware-config/index.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+subgroup: General Resources
+menu_title: config.php settings
+menu_order: 45
+---
+ 
+<div class="toc-list"></div>
+
+## Introduction
+In this guide we will take a closer look at the configuration file `config.php`.
+The file is in the root folder of a shopware installation. Normally it is auto
```
Comment (mitelg) [inline source/developers-guide/shopware-config/index.md]: This file is ... Normally it is generated ...

File: source/developers-guide/shopware-config/index.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+<div class="toc-list"></div>
+
+## Introduction
+In this guide we will take a closer look at the configuration file `config.php`.
+The file is in the root folder of a shopware installation. Normally it is auto 
+generated during the installation prozess and filled with your database credentials. 
+It should look like this:
+```
+<?php
+return array(
```
Comment (bcremer) [inline source/developers-guide/shopware-config/index.md]: Please use the short array syntax everywhere for consistency.

File: source/developers-guide/shopware-config/index.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    'csrfProtection' => [
+        'frontend' => true,
+        'backend' => true
+    ],
+```
+With these options you can activate/deactivate the CSRF attack Protection. By default, both options are set 
+to `true`. Deactivating them is for example necessary if you want to run mink tests 
+with behat. For more information take a look at the complete guide: [CSRF Protection](/developers-guide/csrf-protection/)
+
+### phpsettings
```
Comment (bcremer) [inline source/developers-guide/shopware-config/index.md]: PHP Settings

File: source/developers-guide/shopware-config/index.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+);
+```
+During this guide I will point out some important options and explain them.  
+For a complete list of options you can look at the `engine/Shopware/Configs/Default.php` file 
+which holds all the possible configuration options and their default values. You only
+need to specify options in your `config.php` if you want to override the defaults.  
+But keep in mind that most of these options should only be used for __debugging and testing__ 
+and should be removed for your live system.
+
+## Some important options
```
Comment (bcremer) [inline source/developers-guide/shopware-config/index.md]: Important options

File: source/developers-guide/shopware-config/index.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+        'password' => 'yourPassword',
+        'dbname' => 'yourDbname',
+        'host' => 'yourHost',
+        'port' => 'yourPort'
+    ),
+);
+```
+During this guide I will point out some important options and explain them.  
+For a complete list of options you can look at the `engine/Shopware/Configs/Default.php` file 
+which holds all the possible configuration options and their default values. You only
```
Comment (mitelg) [inline source/developers-guide/shopware-config/index.md]: ... holds all possible...

File: source/developers-guide/shopware-config/index.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+        ...
+        'forceCompile' => false,
+        ...
+    ],
+```
+This option controlls the template caching. Normally you have to clear your Cache after 
+every change on the template, but if you set `forceCompile` to `true` your template will be
+compiled on every reload. This should be an essential option for every developer.
+
+### http cache
```
Comment (bcremer) [inline source/developers-guide/shopware-config/index.md]: HTTP Cache

File: source/developers-guide/shopware-config/index.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+## Some important options
+### CSRF Protection
+default:
+```
+    'csrfProtection' => [
+        'frontend' => true,
+        'backend' => true
+    ],
+```
+With these options you can activate/deactivate the CSRF attack Protection. By default, both options are set
```
Comment (mitelg) [inline source/developers-guide/shopware-config/index.md]: ...attack protection.

File: source/developers-guide/shopware-config/index.md [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+### template
+default:
+```
+    'template' => [
+        ...
+        'forceCompile' => false,
+        ...
+    ],
+```
+This option controlls the template caching. Normally you have to clear your Cache after
```
Comment (mitelg) [inline source/developers-guide/shopware-config/index.md]: ... controls the smarty template caching. ... your cache after ...

File: source/developers-guide/shopware-config/index.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+    'db' => [
+        'username' => 'yourUsername',
+        'password' => 'yourPassword',
+        'dbname' => 'yourDbname',
+        'host' => 'yourHost',
+        'port' => 'yourPort'
+    ],
+];
+```
+During this guide I will point out some important options and explain them.
```
Comment (MarcelSchmaeing) [inline source/developers-guide/shopware-config/index.md]: "I will point out" sounds really like a blog post we should write this more as a reference documentation

**Pull request merged**